tephra-core/src/main/java/org/apache/tephra/TransactionManager.java

```diff
@@ -206,12 +210,19 @@ public TransactionManager(Configuration conf, @Nonnull TransactionStateStorage p
     // TODO: REMOVE WITH txnBackwardsCompatCheck()
     longTimeoutTolerance = conf.getLong("data.tx.long.timeout.tolerance", 10000);
 
-    //
+    ClientIdRetention retention = ClientIdRetention.valueOf(
```

Comment:
Hmmm. We don't do that for the other configurations... if any of the numeric properties cannot be parsed as a number, it also fails. I think it is a good idea to fail on invalid configuration, because if there is a configuration that is present, then that is most likely with the intention to change the default. If there is a typo or some other error, and we only log a warning, that warning is likely to go unnoticed and the system will behave in a way that was intended, and that will go undetected until it causes failures.